### DISCUSSION TITLE
How can I use Pyinstaller to built for macOS Monterey when I'm on macOS Ventura?

### DISCUSSION THREAD

**Discussion by alishobeiri** [pyinstaller/pyinstaller#7830] [Help]
Hi, 

I have been trying to create an executable from my Python script, when I create the executable and run it on macOS Ventura, I have no problems. 

However, if I run it on a device with macOS Monterey I get the following error:
```
dlopen: dlopen(/var/folders/n9/bbttg62x4kvdfhd5z2v880940000gn/T/_MEIZg9KXJ/Python, 0x000A): Symbol not found: (_mkfifoat)
  Referenced from: '/private/var/folders/n9/bbttg62x4kvdfhd5z2v880940000gn/T/_MEIZg9KXJ/Python'
  Expected in: '/usr/lib/libSystem.B.dylib'
```

From what I understand, this issue might be happening because I have built the project on macOS Ventura and so it is not backward compatible. Is there a way to make it backward compatible? 

I tried to install Monterey on a VM, and that did not work (because my CPU cannot support Monterey). Is there any way i can do backward compatibility without needing to get a new machine?

**Comment by alishobeiri** [comment]
I had also seen this, but I couldn't quite make sense of it (if anyone has any tips):
https://gist.github.com/phfaist/a5b8a895b003822df5397731f4673042

**Comment by rokm** [comment]
How did you install python?

The problem is that *python* that you are using was built for Ventura, and is not compatible with older versions of macOS (i.e., it is using a symbol from system library that does not exist on earlier versions). And because PyInstaller collects python shared library from your python installation, your frozen application shares the same limitation.

If you want compatibility with older macOS versions, you need to use compatible python build, for example the official python.org build.

**Comment by alishobeiri** [comment]
Thanks so much for the reply, I am using the Python from a virtualenv, but I haven't verified which Python that is built for. I will make sure to double check that now. 

Thanks a lot for all your help! I will keep you posted on how it goes :)

**Comment by alishobeiri** [comment]
Do you by chance know if there's a way I can check if the built version works for an earlier version without having access to a computer on that earlier version?

**Reply by bwoodsend** [reply]
`otool -L some-file.dylib | grep -i minos` gives you the deployment target for a given binary. If the number reported by that command is less than or equal to 12 for every binary file in you application, you should be fairly safe to run on macOS 12.x.
